scripts/src/java/org/oppia/android/scripts/RegexPatternValidationCheck.kt

```diff
@@ -0,0 +1,232 @@
+
+      var scriptFailedFlag = false
+
+      getFilenameChecks().forEach {
+        if (checkProhibitedFileNamePattern(
+            repoPath = repoPath,
+            searchFiles = searchFiles,
+            prohibitedFilenameRegexString = it.getProhibitedFilenameRegex(),
+            errorToShow = it.getFailureMessage()
+          )) {
+          scriptFailedFlag = true
+        }
+      }
+
+      getFileContentChecks().forEach {
+        if (checkProhibitedContent(
+            repoPath = repoPath,
+            searchFiles = searchFiles,
+            fileNameRegexString = it.getFilenameRegex(),
+            prohibitedContentRegexString = it.getProhibitedContentRegex(),
+            errorToShow = it.getFailureMessage()
+          )) {
+          scriptFailedFlag = true
+        }
+      }
+
+      if (scriptFailedFlag) {
+        throw Exception(ScriptResultConstants.REGEX_CHECKS_FAILED)
+      } else {
+        println(ScriptResultConstants.REGEX_CHECKS_PASSED)
+      }
+    }
+
+    /**
+     * Fetches all filename checks
+     *
+     * @return [List<FilenameCheck>] a list of all the FilenameChecks
+     */
+    fun getFilenameChecks(): List<FilenameCheck> {
+      val fileNamePatternsBinaryFile =
+        File("scripts/assets/filename_pattern_validation_checks.pb")
+      val filenameCheckBuilder = FilenameChecks.newBuilder()
+      val namePatternsObj: FilenameChecks =
+        FileInputStream(fileNamePatternsBinaryFile).use {
+          filenameCheckBuilder.mergeFrom(it)
+        }.build() as FilenameChecks
+
+      return namePatternsObj.getFilenameChecksList()
+    }
+
+    /**
+     * Fetches all file content checks
+     *
+     * @return [List<FileContentCheck>] a list of all the FileContentChecks
+     */
+    fun getFileContentChecks(): List<FileContentCheck> {
+      val fileContentsBinaryFile =
+        File("scripts/assets/file_content_validation_checks.pb")
+      val fileContentCheckBuilder = FileContentChecks.newBuilder()
+      val fileContentsObj: FileContentChecks =
+        FileInputStream(fileContentsBinaryFile).use {
+          fileContentCheckBuilder.mergeFrom(it)
+        }.build() as FileContentChecks
+
+      return fileContentsObj.getFileContentChecksList()
+    }
+
+    /**
+     * Checks for a prohibited file naming pattern
+     *
+     * @param repoPath the path of the repo.
+     * @param searchFiles a list of all the files which needs to be checked.
+     * @param fileNameRegexString filename pattern regex which should not be present
+     * @param errorToShow error to show incase of failure
+     * @return [Boolean] check failed or passed
+     */
+    fun checkProhibitedFileNamePattern(
+      repoPath: String,
+      searchFiles: Sequence<File>,
+      prohibitedFilenameRegexString: String,
+      errorToShow: String
+    ): Boolean {
+      var filenamePatternCheckFailedFlag = false
+      val prohibitedFilenameRegex = prohibitedFilenameRegexString.toRegex()
+      searchFiles.forEach {
```

Comment:
I suggest also using fold here & elsewhere to avoid vars.

Alternatively, you could try something like this:

```kotlin
val matchedFiles = searchFiles.filter { file ->
  val filePath = it.toString().removePrefix(repoPath)
  return@filter prohibitedFilenameRegex.matches(filePath)
}
// Now, log the failure for each matched file, or combined all matched files into one list for a single error message.
```

This might be an interesting approach for here & below since it avoids repeating the same error message over and over. WDYT?

Reply:
This seems neat for the Filename checks, but my only concern is for the FileContent checks. As, we also have to log the line number, so it will be best, if we log its failure as soon as we encounter it.